OPAL/br/src/main/scala/org/opalj/br/analyses/cg/EntryPointFinder.scala

```diff
@@ -347,4 +347,79 @@ object AllEntryPointsFinder extends EntryPointFinder {
             project.allProjectClassFiles.flatMap(_.methodsWithBody)
         else project.allMethodsWithBody
     }
-}
\ No newline at end of file
+}
+
+/**
+ * The AndroidEntryPointFinder considers specific methods of app components as entry points.
+ *
+ * @author Tom Nikisch
+ */
+object AndroidEntryPointsFinder extends EntryPointFinder {
+    override def collectEntryPoints(project: SomeProject): Traversable[Method] = {
+        val classHierarchy = project.classHierarchy
+        val eps = ArrayBuffer.empty[Method]
+
+        var peps: List[String] = List("onCreate", "onRestart", "onStart", "onResume", "onStop", "onDestroy",
+            "onActivityResult")
+        classHierarchy.foreachSubclass(ObjectType("android/app/Activity"), project) { sc ⇒
+            for (pep ← peps) {
+                val mc = sc.findMethod(pep)
+                for (m ← mc) {
+                    if (m.body != null) {
+                        eps.append(m)
+                    }
+                }
+            }
+        }
+
+        peps = List("onCreate", "onStartCommand", "onBind", "onStart")
+        classHierarchy.foreachSubclass(ObjectType("android/app/Service"), project) { sc ⇒
+            for (pep ← peps) {
+                val mc = sc.findMethod(pep)
+                for (m ← mc) {
+                    if (m.body != null) {
+                        eps.append(m)
+                    }
+                }
+            }
+        }
+
+        peps = List("onCreate", "query", "insert", "update")
+        classHierarchy.foreachSubclass(ObjectType("android/content/ContentProvider"), project) { sc ⇒
+            for (pep ← peps) {
+                val mc = sc.findMethod(pep)
+                for (m ← mc) {
+                    if (m.body != null) {
+                        eps.append(m)
+                    }
+                }
+            }
+        }
+
+        peps = List("onLocationChanged", "onProviderDisabled", "onProviderEnabled", "onStatusChanged")
+        classHierarchy.foreachSubtype(ObjectType("android/location/LocationListener")) { st ⇒
```

Comment:
Why are you using foreachSubtype here, but foreachSubclass on the previous ones? foreachSubclass directly provides you with the classfile, already ensuring that it is available.